### PR TITLE
More accurate ldap configuration properties validation

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Prevents some NPE's or unexpected behavior when not all required LDAP properties are set by
throwing appropriate exceptions in those cases.

### CODE REVIEW

File: wsmaster/codenvy-ldap/src/main/java/com/codenvy/ldap/sync/UserMapper.java [hunk 1]
```diff
@@ -40,8 +41,19 @@ public UserMapper(String userIdAttr, String userNameAttr, String userEmailAttr)
...
     public UserImpl apply(LdapEntry entry) {
+        validateRequiredAttrsExists(entry);
         return new UserImpl(entry.getAttribute(idAttr).getStringValue(),
                             entry.getAttribute(mailAttr).getStringValue(),
                             entry.getAttribute(nameAttr).getStringValue());
     }
+
+    private void validateRequiredAttrsExists(LdapEntry entry) {
+        try {
+            requireNonNull(entry.getAttribute(idAttr), "Cannot identify required ID attribute in LDAP entry. Please, check configuration parameter correctness.");
```
Comment (skabashnyuk) [inline wsmaster/codenvy-ldap/src/main/java/com/codenvy/ldap/sync/UserMapper.java]: add parameter name to the message

File: wsmaster/codenvy-ldap/src/main/java/com/codenvy/ldap/auth/AuthenticatorProvider.java [hunk 2]
```diff
@@ -180,7 +205,24 @@ private PooledBindAuthenticationHandler getPooledBindAuthenticationHandler(Poole
...
         final PooledCompareAuthenticationHandler handler = new PooledCompareAuthenticationHandler(
                 connFactory);
+        checkRequiredPropertySet(USER_PASSWORD_ATTRIBUTE_PROPERTY_NAME, userPasswordAttribute);
         handler.setPasswordAttribute(userPasswordAttribute);
         return handler;
     }
+
+    @SafeVarargs
+    @VisibleForTesting
+    final void checkRequiredPropertySet(String name, String value, Pair<String, String>... nameValuePairs) {
```
Comment (skabashnyuk) [inline wsmaster/codenvy-ldap/src/main/java/com/codenvy/ldap/auth/AuthenticatorProvider.java]: may be two methods?

File: wsmaster/codenvy-ldap/src/main/java/com/codenvy/ldap/sync/ProfileMapper.java [hunk 3]
```diff
@@ -44,6 +44,7 @@ public ProfileMapper(String idAttr, Pair<String, String>[] attributes) {
             this.appToLdapAttrNames = ImmutableMap.of();
         } else {
             this.appToLdapAttrNames = ImmutableMap.copyOf(Arrays.stream(attributes)
+                                                                .filter(p -> !p.first.isEmpty() && p.second != null)
```
Comment (evoevodin) [inline wsmaster/codenvy-ldap/src/main/java/com/codenvy/ldap/sync/ProfileMapper.java]: AFAIK you can't get NULL injected where array of string pairs is injected, i think this check 
is useless and shouldn't be present here.

**Comment by evoevodin** [comment]
@mshaposhnik please add yourself as author to the classes you changed

**Pull request merged**